p2p/listener.go

```diff
@@ -115,30 +112,31 @@ func (l *ListenHandler) handleStream(stream network.Stream) {
 			rw.Write(packed)
 			rw.Flush()
 		case "B": // get packed block
-			if 1 != len(parameters) {
-				err = fault.MissingParameters
-				util.LogError(log, util.CoRed, fmt.Sprintf("-->Block length is not equal 1 , length=%d", len(parameters)))
-			} else if 8 == len(parameters[0]) { //it 8 or 6 ??
-				result := storage.Pool.Blocks.Get(parameters[0])
-				if nil == result {
-					err = fault.BlockNotFound
-					listenerSendError(rw, nodeChain, err, "-->Query Block: block not found", log)
-				}
-				respParams := [][]byte{result}
-				packed, err := PackP2PMessage(nodeChain, "B", respParams)
-				if err != nil {
-					listenerSendError(rw, nodeChain, err, "-->Query Block  Information", log)
-					break
+			if 1 == len(parameters) {
+				listenerSendError(rw, nodeChain, fault.MissingParameters, "-->Query Block: invalid parameter", log)
+				if 8 == len(parameters[0]) { //it 8 or 6 ??
```

Comment:
This `if` statement should be in the same level to `1 == len(parameters)`.

Reply:
b4, it use 1!=len(parameters) so it is in the same level. 
I try to use the exclusive way to write it 1==len(parameter) so it is in the same level.
but listenerSendError should not be here but in the buttom

Reply:
OK. Understand